library/src/main/java/net/obsidianx/chakra/Flexbox.kt

```diff
@@ -390,3 +404,35 @@ fun Flexbox(
         }
     }
 }
+
+/**
+ * Get constraints to use for measuring the child node.
+ * This ensures the upoer bound for the cross-axis is not lost, if it was ever present,
+ * when flex box scopes are nested.
+ */
+private fun getMaxConstraintForIntrinsicMinMeasure(
+    childNodeData: FlexNodeData,
+    constraints: Constraints
+): IntrinsicMinMeasurementMaxConstraints {
+    val width = if (childNodeData.isContainer
+        && (childNodeData.style.flexDirection == YogaFlexDirection.COLUMN
+                || childNodeData.style.flexDirection == YogaFlexDirection.COLUMN_REVERSE)
+    ) constraints.maxWidth else Constraints.Infinity
+    val height = if(childNodeData.isContainer && (childNodeData.style.flexDirection == YogaFlexDirection.ROW
+                || childNodeData.style.flexDirection == YogaFlexDirection.ROW_REVERSE)) constraints.maxHeight else Constraints.Infinity
+    return IntrinsicMinMeasurementMaxConstraints(maxWidth = width, maxHeight = height)
+}
+
+private fun clearParentNodeIfNecessary(childNode: YogaNode) {
```

Comment:
This is a preventative measure to guard against a crash I observed a few times before and encountered when messing around with this investigation.
We can't assign a new parent to a child if it already has one.

Don't know how to trigger this by demand normally, but I have seen is sporadically before. Not often though.